pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -136,3 +141,74 @@ func handleAttestCertDelete(ctxArg interface{}, key string,
 	log.Infof("handleAttestCertDelete for %s", status.Key())
 	return
 }
+
+// Run a task certificate post task, on change trigger
+func certsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
+	iteration := 0
+	log.Infoln("starting certs publish task")
+	publishCerts(ctx, iteration)
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(25 * time.Second)
+	agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+
+	for {
+		select {
+		case <-triggerCerts:
+			start := time.Now()
+			iteration++
+			publishCerts(ctx, iteration)
+			pubsub.CheckMaxTimeTopic(agentName+"certs", "publishCerts", start,
+				warningTime, errorTime)
+
+		case <-stillRunning.C:
+		}
+		agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+	}
+}
+
+// prepare the certs list proto message
+func publishCerts(ctx *zedagentContext, iteration int) {
+	var attestReq = &attest.ZAttestReq{}
+
+	startPubTime := time.Now()
+	attestReq.ReqType = attest.ZAttestReqType_ATTEST_REQ_CERT
+	// no quotes
+
+	// TBD:XXX, get the ECDH Certs here
+
+	log.Debugf("publishCerts to ZedCloud, sending %s", attestReq)
+	sendCertsProtobuf(attestReq, iteration)
+	log.Debugf("publishCerts: after send, total elapse sec %v",
+		time.Since(startPubTime).Seconds())
+}
+
+// Try all (first free, then rest) until it gets through.
+// Each iteration we try a different port for load spreading.
+// For each port we try all its local IP addresses until we get a success.
+func sendCertsProtobuf(attestReq *attest.ZAttestReq, iteration int) {
+	data, err := proto.Marshal(attestReq)
+	if err != nil {
+		log.Fatal("SendInfoProtobufStr proto marshaling error: ", err)
+	}
+
+	buf := bytes.NewBuffer(data)
+	size := int64(proto.Size(attestReq))
+	attestURL := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API,
+		false, devUUID, "attest")
+	const return400 = false
+	_, _, rtf, err := zedcloud.SendOnAllIntf(&zedcloudCtx, attestURL,
+		size, buf, iteration, return400)
+	if err != nil {
+		// Hopefully next timeout will be more successful
```

Comment:
There are no timeouts; AFAICT this message needs to you the defer logic to make sure it gets resent after a failure. And the defer logic (SetDeferred) needs a unique key for the message; I guess any string would suffice since there is only ever one cert message the device tries to send.

Reply:
Have added the base logic, without any presumption about the content. Will refine more with the detailed functional pull request, which will be using this pull requwest as base.